### PR TITLE
Improve AzMon Trace Exporter support for EventHubs

### PR DESCRIPTION
**Pull request by trask**
The exporter needs to apply messaging semantic conventions to EventHubs spans.

Ideally EventHubs SDK should conform to the messaging semantic conventions, but no rush as those are still not finalized and may still change.

### CODE REVIEW

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorTraceExporter.java [hunk 1]
```diff
@@ -262,11 +265,17 @@ private void applySemanticConventions(SpanData span, RemoteDependencyData remote
...
             applyMessagingClientSpan(attributes, remoteDependencyData, messagingSystem, span.getKind());
             return;
         }
+        // TODO (trask) ideally EventHubs SDK should conform and fit the above path used for other messaging systems
+        //  but no rush as messaging semantic conventions may still change
+        String name = span.getName();
+        if (name.equals("EventHubs.send") || name.equals("EventHubs.message")) {
+            applyEventHubsSpan(attributes, remoteDependencyData);
+            return;
+        }
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorTraceExporter.java]: A similar change should be made for Service Bus too.

I have created a couple of issues to track this for Event Hubs and Service Bus. 
https://github.com/Azure/azure-sdk-for-java/issues/21684
https://github.com/Azure/azure-sdk-for-java/issues/21686

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorTraceExporter.java [hunk 2]
```diff
@@ -404,6 +413,15 @@ private void applyMessagingClientSpan(Attributes attributes, RemoteDependencyDat
         }
     }
 
+    // TODO (trask) ideally EventHubs SDK should conform and fit the above path used for other messaging systems
+    //  but no rush as messaging semantic conventions may still change
+    private void applyEventHubsSpan(Attributes attributes, RemoteDependencyData telemetry) {
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorTraceExporter.java:418]: Another method should be added to handle Service Bus spans as well.

**Comment by lmolkova** [comment]
Based on later discussions, we're going to stick to the current spec until we update messaging OTel conventions to support EventHubs scenarios - we're working on it with AzMon team.
We're [making the current spec](https://gist.github.com/lmolkova/e4215c0f44a49ef824983382762e6b92) more official and adding [mappings to common-schema](https://github.com/microsoft/common-schema/blob/main/Mappings/AzureSDKs.md). This is a temporary solution and we'll be converging to OTel conventions once they mature - this is also across 4 languages.

The concern here is that OTel conventions are experimental and might change and we'd have to diverge or follow them.

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorTraceExporter.java [hunk 3]
```diff
@@ -264,7 +269,22 @@ private void applySemanticConventions(Attributes attributes, RemoteDependencyDat
...
+        //  https://github.com/Azure/azure-sdk-for-java/issues/21684
+        String name = span.getName();
+        if (name.equals("EventHubs.send") || name.equals("EventHubs.message")) {
+            applyEventHubsSpan(attributes, remoteDependencyData);
+            return;
+        }
+        // TODO (trask) ideally ServiceBus SDK should conform and fit the above path used for other messaging systems
+        //  but no rush as messaging semantic conventions may still change
+        //  https://github.com/Azure/azure-sdk-for-java/issues/21686
+        if (name.equals("ServiceBus.message") || name.equals("ServiceBus.process")) {
```
Comment (lmolkova) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorTraceExporter.java:286]: `ServiceBus.process` is a consumer span and should be request telemetry. Perhaps this should be `ServiceBus.send`?

**Review by lmolkova — COMMENTED** [review]
What about request telemtery support? https://gist.github.com/lmolkova/e4215c0f44a49ef824983382762e6b92#file-z_azure_monitor_exporter_mapping-md

It was important for some portal experiences.

**Comment by trask** [comment]
> What about request telemetry support? https://gist.github.com/lmolkova/e4215c0f44a49ef824983382762e6b92#file-z_azure_monitor_exporter_mapping-md
> 
> It was important for some portal experiences.
> [edit] : i guess `customMeasurements["timeSinceEnqueued"]` is irrelevant without metrics support, but `source` is still useful - it allows consumer to now which EventHubs it interacted with

oh nice! this is done now

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorTraceExporter.java [hunk 4]
```diff
@@ -404,6 +403,24 @@ private void applyMessagingClientSpan(Attributes attributes, RemoteDependencyDat
...
 
+    // special case needed until Azure SDK moves to OTel semantic conventions
+    private static void applyEventHubsSpan(Attributes attributes, RemoteDependencyData telemetry) {
+        telemetry.setType("Microsoft.EventHub");
+        telemetry.setTarget(getAzureSdkTargetSource(attributes));
+    }
+
+    // special case needed until Azure SDK moves to OTel semantic conventions
+    private static void applyServiceBusSpan(Attributes attributes, RemoteDependencyData telemetry) {
+        telemetry.setType("AZURE SERVICE BUS");
```
[the remark this answers is not in the document]
Comment (trask) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorTraceExporter.java:414]: ya, i'll raise this with U/X team to support `Microsoft.ServiceBus` 👍

**Pull request merged**